src/publicHolidayReminder.ts

```diff
@@ -41,33 +48,53 @@ const toDateStamp = date => {
 //       method: 'PUBLISH'
 //   },
 //
-const findRelevantEvents = (rawEvents, date) => {
-  return Object.values(rawEvents)
-    .filter(rawEvent => toDateStamp(rawEvent.start) === toDateStamp(date))
+const findRelevantEvents = (
+  calendarResponse: CalendarResponse,
+  date: Date
+): MyEvent[] => {
+  return Object.values(calendarResponse)
+    .filter(rawEvent => {
+      return (
+        isVEvent(rawEvent) && toDateStamp(rawEvent.start) === toDateStamp(date)
+      );
+    })
     .map(rawEvent => {
+      const vEvent = rawEvent as VEvent; // safe to cast here because we ran isVEvent in filter() above
+
       return {
-        start: rawEvent.start,
-        description: `${rawEvent.description} (${rawEvent.summary})`
+        start: vEvent.start,
+        description: `${vEvent.description} (${vEvent.summary})`
       };
     });
 };
 
-const getCalendarData = async filePathOrUrl => {
+const isVEvent = (item: CalendarComponent): item is VEvent => {
+  return item.type === 'VEVENT';
+};
+
+interface MyEvent {
+  start: DateWithTimeZone;
+  description: string;
+}
```

Comment:
I put most types in `types.ts` but for some types which are only used within a single module I have left them in the module. Not sure if this is the right call.

Reply:
imo the important things are to encapsulate the types based on their namespace/coupling, and that you don't create unneeded import/exports by pulling out every type even if it's only being used by one module.

For this project I think what you've got is pretty fine, since I don't think there's enough actual code to justify any extra namespaces but if there was a bit more to it I'd say you should maybe think about having `projectworks` and `notifier` namespaces in which case their types related types should live in there too.

Reply:
Thanks. Good to know where to go when/if we add more code